form-regex.rst

```diff
@@ -0,0 +1,101 @@
+    - The question mark indicates zero or one occurrence of the preceding element. For example, ``colou?r`` matches both color and colour.
+    - The asterisk indicates zero or more occurrences of the preceding element. For example, ``ab*c`` matches ac, abc, abbc, abbbc, and so on.
+    - The plus sign indicates one or more occurrences of the preceding element. For example, ``ab+c`` matches abc, abbc, abbbc, and so on, but not ac.
+    - ``{n}``: The preceding item is matched exactly n times. 
+    - ``{min,}``: The preceding item is matched min or more times. 
+    - ``{min,max}``: The preceding item is matched at least min times, but not more than max times. 
+
+
+For a clear understanding of regular expressions, try these regex online checker tools: 
+
+- https://regex101.com/
+- https://www.regextester.com/
+- https://regexr.com/
+- http://www.regexe.com/.
+
+.. _tips-on-regex:
+
+Tips on using regular expressions
+----------------------------------
+
+- Regular expressions only apply to **strings**. The function used is of the form: ``regex(string value, string expression)`` where it returns the result of regex test on provided value. If the result is true, the input value is valid and satisfies the constraint. If the result is false, input value is not valid and user may need to re-enter the input. The regex function will be placed in the constraints column in your XLSForm. 
+
+- Be careful while setting limits on length of a constraint.
+  For example, if you want to restrict a text input to a maximum of six alphabetic characters, you can do so as follows:
+
+ In **XLSForm**:
+
+ .. csv-table:: **Survey**
+   :header: "type", "name", "label", "constraint", "constraint_message"
+   :widths: auto
+
+   "text", "ex", "Enter example", "regex(.,'^[a-zA-Z]{0,6}$')", "Input can have a maximum of 6 alphabetic characters."
+
+ In **XForm XML**:
+
+ .. code-block:: xml
+
+   <bind constraint="Regex(.,'^[a-zA-Z]{0,6}$')" jr:constraintMsg="Input can have a maximum of 6 alphabetic characters." nodeset="/regex_ex/ex" type="string"/>
+
+   <input ref="/regex_ex/ex">
+      <label>Enter example</label>
+    </input>
+
+ Instead if you used a constraint of the form ``regex(.,'^[a-zA-Z]{6}$')``, it would require an input of exactly six alphabetic characters.
+
+ .. note::
+
+  - ``.`` denotes the input string.
+  - ``^`` and ``$`` denote start and end of string respectively.
+
+- If you want to use a regular expression constraint on a number, first of all, make sure that the type of your question is **text** and appearance is **numbers** and then apply the constraint. 
+
+ The following example will validate a 10-digit North American telephone number. Separators are not required, but can include spaces, hyphens, or periods. Parentheses around the area code are also optional.
```

Comment:
Clever.

Reply:
I actually got the example from the list you provided. :stuck_out_tongue: